Chocolate_Vending_Machine.java

```diff
@@ -0,0 +1,240 @@
+package Week_7;
+
+import java.util.*;
+
+public class Chocolate_Vending_Machine implements Cash_Inventory, Item_Inventory {
+	private HashMap<Coin, Integer> cash = new HashMap<>();
+	HashMap<Item, Integer> items = new HashMap<>();
+
+	public void initialize_CashInventory() {
+	
+		for (Coin c : Coin.values()) {
+			cash.put(c, 5);
+		}
+	}
+
+	public void initialize_CashInventory(int coins[]) {

```

Comment:
this seems a little scary for a public function as there is no guarantee that the coins array has as many elements as Coin.Values potentially leading to an index out of bounds

Reply:
Should this and the above method be public at all?  Do we want anyone to be able to re-initialize the cash inventory or should this be an internal operation

Reply:
same comments about the initialize Item inventories

Reply:
Yes, I need to make it private